src/Bridge/Doctrine/Orm/Extension/FilterExtension.php

```diff
@@ -60,12 +61,25 @@ public function applyToCollection(QueryBuilder $queryBuilder, QueryNameGenerator
             return;
         }
 
+        $orderFilter = null;
+
         foreach ($resourceFilters as $filterId) {
             $filter = $this->getFilter($filterId);
             if ($filter instanceof FilterInterface) {
+                // Apply the OrderFilter after every other filter to avoid an edge case where OrderFilter would do a LEFT JOIN instead of an INNER JOIN
+                if ($filter instanceof OrderFilter) {
```

Comment:
it wouldn't be better to give filters priorities than hard code like this?

Reply:
It may be better but it'd have required more work (we currently have 2 way of instantiating filters), and there is only this case that's impacted by the order. This quick fix is enough for now imo.